client/trino-cli/src/main/java/io/trino/cli/Query.java

```diff
@@ -162,7 +163,7 @@ private boolean renderQueryOutput(Terminal terminal, PrintStream out, PrintStrea
         if (client.isRunning() || (client.isFinished() && client.finalStatusInfo().getError() == null)) {
             QueryStatusInfo results = client.isRunning() ? client.currentStatusInfo() : client.finalStatusInfo();
             if (results.getUpdateType() != null) {
-                renderUpdate(errorChannel, results);
+                renderUpdate(errorChannel, results, outputFormat, usePager);
```

Comment:
Can the server not set the update type in results if it's an explain query?

Reply:
I tried this, see my comment below with link to a pr that has links to 2 others :D I did 3 different approaches but none of them seems good enough